analyses/DESeq_analysis/run-DESeq-analysis.R

```diff
@@ -0,0 +1,195 @@
+
+#Determine the mean TPM of the tissue. If there are multiple samples use the mean TPM
+if(Cancer.Hist_Hits > 1) Histology_MEAN_TPMs = round(apply(TPMData_filtered[match(rownames(Result),rownames(TPMData_filtered)),which(colnames(TPMData_filtered) %in%  sample_type_df_filtered[which(sample_type_df_filtered$Type %in% c(histology_filtered[I])),1])]  ,MARGIN=1, mean ),2)
+
+#Determine the mean TPM of the tissue. If there is one sample just use the single TPM value of the sample
+if(Cancer.Hist_Hits <= 1) Histology_MEAN_TPMs = TPMData_filtered[match(rownames(Result),rownames(TPMData_filtered)),which(colnames(TPMData_filtered) %in%  sample_type_df_filtered[which(sample_type_df_filtered$Type %in% c(histology_filtered[I])),1])]
+
+#Round the mean TPMs to the 2 decimal places
+Histology_MEAN_TPMs = round(Histology_MEAN_TPMs,2)
+GTEX_MEAN_TPMs = round(GTEX_MEAN_TPMs,2)
+
+
+#Create Final Dataframe with all the info calculated and extracted from histology file Including EFO/MONDO codes where available and RMTL status
+Final_Data_Table = data.frame(
+  datasourceId = paste(strsplit(histology_filtered[I],split="_")[[1]][1],"vs_GTex",sep="_"),
+  datatypeId = "rna_expression",
+  cohort = paste(unique(hist$cohort[which(hist$Kids_First_Biospecimen_ID %in% HIST_sample_type_df_filtered$Case_ID)]),collapse=";",sep=";"),
+  gene_symbol = rownames(Result),
+  gene_id = ENSG_Hugo$ensg_id[match(rownames(Result),ENSG_Hugo$gene_symbol)],
+  RMTL = ENSG_Hugo$rmtl[match(rownames(Result),ENSG_Hugo$gene_symbol)],
+  EFO = ifelse(length(which(EFO_MONDO$cancer_group == unique(hist$cancer_group[which(hist$Kids_First_Biospecimen_ID %in% HIST_sample_type_df_filtered$Case_ID)]))) >= 1, EFO_MONDO$efo_code[which(EFO_MONDO$cancer_group == unique(hist$cancer_group[which(hist$Kids_First_Biospecimen_ID %in% HIST_sample_type_df_filtered$Case_ID)]))], "" ),
+  MONDO = ifelse(length(which(EFO_MONDO$cancer_group == unique(hist$cancer_group[which(hist$Kids_First_Biospecimen_ID %in% HIST_sample_type_df_filtered$Case_ID)]))) >= 1,EFO_MONDO$mondo_code[which(EFO_MONDO$cancer_group == unique(hist$cancer_group[which(hist$Kids_First_Biospecimen_ID %in% HIST_sample_type_df_filtered$Case_ID)]))],""),
+  comparisonId = gsub(" |/|;|:|\\(|)","_",paste(histology_filtered[I],GTEX_filtered[J],sep="_v_")),
+  cancer_group = paste(unlist(strsplit(histology_filtered[I],split="_"))[-1],collapse=" "),
+  cancer_group_Count = Cancer.Hist_Hits,
+  GTEx = GTEX_filtered[J],
+  GTEx_Count = GTEX_Hits,
+  cancer_group_MeanTpm = Histology_MEAN_TPMs,
+  GTEx_MeanTpm = GTEX_MEAN_TPMs,
+  Result
+)#Final_Data_Table = data.frame(
+
+
+#Save files
+system("mkdir Results/")
+
+#Define file name as Histoloy_v_Gtex.tsv and replacing all 'special symbols' with '_' for the filename
+FILENAME <- gsub(" |/|;|:|\\(|)","_",paste(histology_filtered[I],GTEX_filtered[J],sep="_v_"))
+write.table(Final_Data_Table,file=paste("Results_5_forv6/",FILENAME,".tsv",sep=""),sep="\t",col.names = T, row.names = F,quote = F)
```

Comment:
To be consistent with the created directory name.

```suggestion
write.table(Final_Data_Table,file=paste("Results/",FILENAME,".tsv",sep=""),sep="\t",col.names = T, row.names = F,quote = F)
```